internal/algod/state.go

```diff
@@ -194,9 +194,18 @@ func (s *StateModel) UpdateKeys(ctx context.Context, t system.Time) {
 			if err != nil {
 				continue
 			}
-
 			s.Accounts[acct.Address] = s.Accounts[acct.Address].Merge(rpcAcct)
 			s.Accounts[acct.Address] = s.Accounts[acct.Address].UpdateExpiredTime(t, s.ParticipationKeys, int(s.Status.LastRound), s.Metrics.RoundTime)
+
+			// Check if the account is online but expired,
+			//can happen when a node is offline when the key expired
+			if rpcAcct.Status == "Online" && s.Accounts[acct.Address].Expires != nil {
+				if s.Accounts[acct.Address].Expires.Before(time.Now()) {
+					addr := s.Accounts[acct.Address]
+					addr.Status = "Offline"
+					s.Accounts[acct.Address] = addr
+				}
+			}
```

Comment:
```suggestion
			if rpcAcct.Status == "Online" && s.Status.LastRound > 0 && uint64(rpcAcct.Participation.VoteLastValid) < s.Status.LastRound {
				addr := s.Accounts[acct.Address]
				addr.Status = "Offline"
				s.Accounts[acct.Address] = addr
			}
```
I think it would be better to compare the actual last-vote-round here instead of using expiry time

(Not sure if there should be more guards here e.g. in account.participation)